src/v2/ion/ViewClassNamesFactory.js

```diff
@@ -0,0 +1,88 @@
+/* eslint complexity: [2, 25] */
+import {FORMS} from './RemediationConstants';
+
+const FORMNAME_CLASSNAME_MAPPINGS = {
+  [FORMS.IDENTIFY]: {
+    'identify': ['identify', 'primary-auth'],
+    'password': ['identify', 'primary-auth']
+  },
+  [FORMS.ENROLL_PROFILE]: {
+    'enroll-profile': ['enroll-profile', 'registration'],
+  },
+  [FORMS.CHALLENGE_AUTHENTICATOR]: {
+    email: ['challenge-authenticator-email', 'mfa-verify-passcode'],
+    password: ['challenge-authenticator-password', 'mfa-verify-password'],
+    sms: ['challenge-authenticator-voice', 'mfa-verify-passcode'],
+    voice: ['challenge-authenticator-voice', 'mfa-verify-passcode'],
+    'security_question': ['challenge-authenticator-security_question', 'mfa-verify-question']
+  },
+  [FORMS.ENROLL_AUTHENTICATOR]: {
+    email: ['enroll-authenticator-email', 'enroll-email'],
+    password: ['enroll-authenticator-password', 'enroll-password'],
+    sms: ['enroll-authenticator-sms', 'enroll-sms'],
+    voice: ['challenge-authenticator-voice', 'mfa-verify-passcode'],
+    'security_question': ['enroll-authenticator-security_question', 'enroll-question'],
+  },
+  [FORMS.AUTHENTICATOR_ENROLLMENT_DATA]: {
+    voice: ['authenticator-enrollment-data'],
+    sms: ['authenticator-enrollment-data']
+  },
+  [FORMS.AUTHENTICATOR_VERIFICATION_DATA]: {
+    voice: ['authenticator-verification-data'],
+    sms: ['authenticator-verification-data']
+  },
+  [FORMS.SELECT_AUTHENTICATOR_ENROLL]: {
+    'select-authenticator-enroll': ['select-authenticator-enroll', 'enroll-choices']
+  },
+  [FORMS.SELECT_AUTHENTICATOR_AUTHENTICATE]: {
+    'password': ['select-authenticator-authenticate']
+  },
+  [FORMS.REENROLL_AUTHENTICATOR]: {
+    'password':['reenroll-authenticator', 'password-expired']
+  },
+  [FORMS.REENROLL_AUTHENTICATOR_WARNING]: {
+    'password':['reenroll-authenticator_warning']
+  },
+  [FORMS.RESET_AUTHENTICATOR]: {
+    'password':['reset-authenticator', 'forgot-password']
+  },
+};
+
+const getClassNameMapping = (appState) => {
+  const formName = appState.get('currentFormName');
+  const authenticator = appState.get('currentAuthenticatorEnrollment') || appState.get('currentAuthenticator');
+  let authenticatorType = authenticator && authenticator.type;
```

Comment:
`appState.get('authenticatorType')`

Reply:
Referencing the derived property from appState actually exposing a bug we have where the derived properties are not getting updated correctly when the ION response changes. When we clear the appState, it still keeps the values for the derived properties intact causing randomness.

Reply:
that sounds big trouble. 
when we render the view, AppState **shall** been refreshed. (last two line in `AppState#setIonResponse`)
Otherwise we need to figure out why/how the bug happens.

Reply:
I have a feeling its similar to what @magizh-okta had once brought up related to derived properties https://okta.slack.com/archives/C7KV3NLE8/p1588011981249400

Reply:
It's slightly different. Magizh was listening to a derived prop changes and in the handler he expects all derived props got updated. however it's not the case yet (courage bug) because derived prop update (plus fire change event) is happening in sequence.
At your case, when we hit `className` function where invokes `getClassNameMapping`, all derived props shall have been updated. Otherwise it's another bug or something I wasn't aware.

Reply:
clearing the app state cache fixed this issue this.trigger('cache:clear');